cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -49,6 +49,30 @@ pub enum Inst {
         dst: Writable<Reg>,
     },
 
+    /// Integer quotient and remainder: (div idiv) $rax $rdx (reg addr)
+    Div {
+        size: u8, // 1, 2, 4 or 8
```

Comment:
Perhaps we should have an enum for this, like the `InstSize` enum over in aarch64? (We could make it `#[repr(u8)]` and assign the arms the above numeric values.)

Reply:
Yep, it'll probably be part of a greater refactoring though; some instructions use a boolean to distinguish 32 bits encodings from 64 bits encodings, not implementing all the other ones. Maybe cg_clif wants the smaller sizes encodings? But if not, implementing only 32 and 64 bits might be sufficient in general.

Reply:
As a general comment, I think we should try to avoid generating 8- and 16- bit instructions as much as possible, and instead generate 32-bit insns where we ignore the upper 24 or 16 bits of the result.  The reason is to avoid partial register stalls.  These are documented in the Intel Opt manual, in my copy Sec 3.5.2.4 (Partial Register Stalls), with similar comments in following sections (Partial XMM Register Stalls, Partial Flag Register Stalls).

Also, 16-bit instructions which contain an immediate field are subject to the stalls described in Sec 3.4.2.3 (Length Changing Prefixes).